src/basic/fd-util.c

```diff
@@ -864,3 +865,52 @@ int fd_get_diskseq(int fd, uint64_t *ret) {
 
         return 0;
 }
+
+int dir_fd_is_root(int dir_fd) {
+        STRUCT_NEW_STATX_DEFINE(st);
+        STRUCT_NEW_STATX_DEFINE(pst);
+        int r;
+
+        assert(dir_fd >= 0);
+
+        r = statx_fallback(dir_fd, "", AT_EMPTY_PATH, STATX_TYPE|STATX_INO|STATX_MNT_ID, &st.sx);
```

Comment:
hmm, path_get_mnt_id() already queries the mnt id via statx() as first attempt? why do this here beforehand?

Reply:
oh, i see now

Reply:
To save on syscalls on the happy path.